ui/app/src/components/PendingApprovalDashlet/PendingApprovalDashlet.tsx

```diff
@@ -164,7 +165,10 @@ export function PendingApprovalDashlet(props: PendingApprovalDashletProps) {
   }, [loadPage, refs]);
 
   useEffect(() => {
-    refs.current.loadPagesUntil(refs.current.currentPage, filterState.selectedTypes);
+    // To avoid re-fetching when it first loads
+    if (refs.current.items) {
```

Comment:
@rart when loading the dashlet, there are two effects that call functions that end up calling the same API. 
loadPage is the one we want to call when loading the dashlet.
loadPagesUntil is used to retrieve all pages until page 'x', useful when reloading the current page, changing a filter in the dashlet (this is the reason why the effect is used) and reloading due to a websocket event. We load all of the pages because there may be changes (for example the number of items is now reduced, so current page is not valid. That's considered in loadPagesUntil).